src/base-apis.js

```diff
@@ -26,6 +29,17 @@ limitations under the License.
 const httpApi = require("./http-api");
 const utils = require("./utils");
 
+function termsUrlForService(serviceType, baseUrl) {
+    switch (serviceType) {
+        case SERVICE_TYPES.IS:
+            return baseUrl + httpApi.PREFIX_IDENTITY_V2 + '/terms';
+        case SERVICE_TYPES.IM:
+            return baseUrl + '/terms';
```

Comment:
:( at not using the namespace for the IM

Reply:
I looked at changing but stopped on thinking about changing all the other ones. I guess we could just use it for this one though.

Reply:
I'd recommend it, just for the sake of not making the problem worse. The `baseUrl` might have a path component (scalar is `/api`, dimension is `/api/v1/scalar`) which might have to be trimmed off before applying the spec prefix + `/terms`

Reply:
Ah yes, the fact that we have configs with the api prefix is the main problem - from a riot/scalar PoV we can just ship the config change with the new app version but I don't know how this would work for anyone with a custom IM: I was going to run it past you with your Dimension hat on.

Reply:
Right, so we can't just update this path because then we'd have to move the `/api/` from the config to the hardcoded part in the source, and this would break dimension since for dimension, the `/api` part goes in the middle.

I think our only option is going to be to update all the paths at once, so once scalar supports `_matrix` paths, we can then patch riot develop to use `_matrix` paths across the board and it will be compatible with deployed scalar and any dimension new enough to support the `_matrix` paths.

Reply:
Dimension supports the route at the root, like the spec implies. You're welcome to copy the parsing from Dimension (which does use the root-level route on upstreams):

```
const parsed = new URL(baseUrl);
parsed.pathname = `/_matrix/integrations/v1/terms`;
return parsed.toString();
```
Source: https://github.com/turt2live/matrix-dimension/blob/9cc1454527fecca90ea7987e9ac15693111c8f32/src/scalar/ScalarClient.ts#L34-L39